### PR TITLE
Adds a bit of terminology documentation.

### PR DESCRIPTION
**Pull request by wohe**
Changes code to match the new documentation.

### CODE REVIEW

File: cartographer/mapping/trajectory_node.h [hunk 1]
```diff
@@ -57,7 +57,7 @@ struct TrajectoryNode {
   // node is being trimmed, it must survive until all use finishes.
   std::shared_ptr<const Data> constant_data;
 
-  transform::Rigid3d pose;
+  transform::Rigid3d global_pose;
```
Comment (cschuet) [inline cartographer/mapping/trajectory_node.h]: // The node pose in the global SLAM frame.

File: docs/source/terminology.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Terminology
+===========
+
+This documents a few common patterns that exist in the Cartographer codebase.
+
+Frames
+======
+
+global (map) frame
+  This is the frame in which global SLAM results are expressed. It is the fixed
```
Comment (cschuet) [inline docs/source/terminology.rst]: What does fixed mean here?

File: docs/source/terminology.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  map frame including all loop closure and optimization results. The transform
+  between this frame and any other frame can jump when new optimization results
+  are available. Its z-axis is pointing upwards.
```
Comment (cschuet) [inline docs/source/terminology.rst]: "upwards..." maybe add "i.e. anti-parallel to the gravitational acceleration vector."

File: docs/source/terminology.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+local (map) frame
+  This is the frame in which local SLAM results are expressed. It is the fixed
```
Comment (cschuet) [inline docs/source/terminology.rst]: What is fixed?

File: docs/source/terminology.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  map frame excluding loop closures and the pose graph optimization. The
+  transform for a given point in time, between this frame and all other frames
+  will never be changed later, the only exception being the global map frame.
```
Comment (cschuet) [inline docs/source/terminology.rst]: This frame also has the z-axis anti-parallel to gravity?

File: docs/source/terminology.rst [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+
+Transforms
+==========
+
+local_pose
+  Transforms data from the tracking frame to the local frame.
+
+global_pose
+  Transforms data from the tracking frame to the global frame.
```
Comment (cschuet) [inline docs/source/terminology.rst]: Add definition of the gravity_alignment vector

gravity_alignment
  Transforms data from the tracking frame to the gravity aligned tracking frame.

File: docs/source/terminology.rst [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+local (map) frame
+  This is the frame in which local SLAM results are expressed. It is the fixed
+  map frame excluding loop closures and the pose graph optimization. The
+  transform for a given point in time, between this frame and all other frames
+  will never be changed later, the only exception being the global map frame.
+
+tracking frame
+  The frame in which all sensor data is expected to be.
+
```
Comment (cschuet) [inline docs/source/terminology.rst]: Add gravity aligned tracking frame

File: docs/source/terminology.rst (hunk 2 above)
Comment (wohe) [reply docs/source/terminology.rst]: It does not move as time elapses.

File: docs/source/terminology.rst (hunk 5 above)
Comment (wohe) [reply docs/source/terminology.rst]: No.

File: docs/source/terminology.rst (hunk 7 above)
Comment (wohe) [reply docs/source/terminology.rst]: I want to keep this PR minimal. The gravity aligned and submap frames can be added in a separate PR.

File: docs/source/terminology.rst (hunk 5 above)
Comment (drigz) [reply docs/source/terminology.rst]: This sentence was a bit hard to understand. Maybe:
"For a given point in time, the transform between this and the global map frame may change, but the transform between this and all other frames does not not change."

File: docs/source/terminology.rst (hunk 7 above)
Comment (drigz) [reply docs/source/terminology.rst]: Maybe just "The frame in which sensor data is expressed."

Comment (drigz) [reply docs/source/terminology.rst]: Also, if you add the gravity-aligned tracking frame, use that hyphen for clarity.

**Pull request merged**